lib/jobs/embedding-generation.ts

```
@@ -0,0 +1,197 @@
+  articleId?: string; // Generate embedding for specific article
+  batchSize?: number; // Batch process articles without embeddings
+  forceRegenerate?: boolean; // Regenerate even if embedding exists
+}
+
+export class EmbeddingGenerationJob {
+  static async process(data: EmbeddingGenerationJobData): Promise<void> {
+    console.log('Processing embedding generation job:', data);
+
+    try {
+      if (data.articleId) {
+        await this.generateSingleEmbedding(data.articleId, data.forceRegenerate);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)